docs/installation/introduction.rst

```diff
@@ -1,6 +1,125 @@
+picked for optimal on-premises deployments, including pre-configured monitoring
+and alerting, self-healing system configuration, and more.
+
+The installation of a MetalK8s cluster can be broken down into
+the following steps:
+
+#. :doc:`Setup <./setup>` of the environment
+#. :doc:`Deployment <./bootstrap>` of the :term:`Bootstrap node`, the first
+   machine in the cluster
+#. :doc:`Expansion <./expansion>` of the cluster, orchestrated from the
+   Bootstrap node
+#. :doc:`Post installation <./post-install>` configuration steps and sanity
+   checks
+
+.. _Kubernetes: https://kubernetes.io/
+
+.. _installation-intro-architecture:
+
+Choosing a Deployment Architecture
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Before starting the installation, choosing an architecture is recommended, as
+it can impact sizing of the machines and other infrastructure-related details.
+
+.. note:: "Machines" may indicate bare-metal servers or VMs interchangeably
+
+.. warning::
+
+   MetalK8s is not designed to handle world-distributed multi-sites
+   architectures. Instead, it focuses on providing a highly resilient cluster
+   at the datacenter scale. To manage multiple sites, look into solutions
+   provided at the application level, or alternatives from the community
+   (such as what `the SIG Multicluster
+   <https://github.com/kubernetes/community/tree/master/sig-multicluster>`_
+   provides).
+
+Standard Architecture
+"""""""""""""""""""""
+The recommended architecture when installing a small size MetalK8s cluster
+emphasizes ease of installation, while providing a high stability for the
+scheduled workloads:
+
+- One machine running Bootstrap and control plane services
+- Two other machines running control plane and Infra services
+- Three more machines for workload applications
+
+.. image:: img/standard-arch.png
+   :width: 100%
+
+Machines dedicated to the control plane do not need large
+amounts of resources (see
+:ref:`the sizing notes below<installation-intro-sizing>`, and can safely run as
+virtual machines. Running workloads on dedicated machines also allows for
+simpler sizing of said machines, as MetalK8s impact would be negligible.
+
+Extended Architecture
+"""""""""""""""""""""
+MetalK8s is designed with on-premise, offline, reliable deployments in mind.
+This example architecture focuses on reliability rather than compacity,
+offering the finest control over the entire platform:
+
+- One machine dedicated to running Bootstrap services (see
+  :ref:`the Bootstrap role<node-role-bootstrap>` definition below)
+- Three extra machines (or five if installing a really large cluster,
+  100+ nodes) for running the Kubernetes_ control plane (with
+  :ref:`core K8s services<node-role-master>` and the backing
+  :ref:`etcd DB<node-role-etcd>`)
+- One or more machines dedicated to running Infra services (see
+  :ref:`the Infra role<node-role-infra>`)
+- Any number of machines dedicated to running applications, the number and
```

Comment:
Any == too vague. Can we give a range? Or suggest a number?

Reply:
No we can't really, it depends on what applications you'll want to run...